test/distributed/_composable/test_composability/test_pp_composability.py

```diff
@@ -58,6 +67,20 @@ def forward(self, x):
         return x
 
 
+class MLPModuleEven(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.net1 = nn.Linear(8, 8)
+        self.net2 = nn.Linear(8, 8)
+        self.net3 = nn.Linear(8, 16)
```

Comment:
To answer your q, here you have to use 16 because of the colwise you apply to net3.